File: algorithms/SumDigits.java
package algorithms;

public class SumDigits {
    public static int sumDigits(int num) {
        while (num > 9) {
            int total = 0;
            while (num != 0) {
                total = total + num % 10;
                num = num / 10;
            }
            num = total;
        }
        return num;
    }

    public static void main(String[] args) {
        int num = 123456;
        int total = sumDigits1(num);
        System.out.println("Sum of digits of number " + num + ": " + total);
        // output: Sum of digits of number 123456: 3
    }

    public static int sumDigits1(int num) {
        return 0;
    }
}

File: algorithmsJS/SumDigits.js
class SumDigits {
  static sumDigits(num) {
    while (num > 9) {
      let total = 0;
      while (num !== 0) {
        total += num % 10;
        num = Math.floor(num / 10);
      }
      num = total;
    }
    return num;
  }

  static sumDigits1(num) {
    return null;
  }
}

// Example usage:
const num = 123456;
const total = SumDigits.sumDigits(num);
const total1 = SumDigits.sumDigits1(num);
console.log(`Sum of digits of number ${num}: ${total}`);
console.log(`Sum of digits of number ${num}: ${total1}`);
